### PR TITLE
 Api gets a random joke from external Api

### PR DESCRIPTION
**Pull request by johanperjulius1**
## This PR contains:
- request spec
- index route
-  JokeGetter controller with index action
- mocked external API response

### CODE REVIEW

File: app/controllers/api/v1/joke_getter_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Api::V1::JokeGetterController < ApplicationController
```
Comment (tochman) [inline app/controllers/api/v1/joke_getter_controller.rb]: If you want to follow convention, use `Api::V1::JokesController`

File: app/controllers/api/v1/joke_getter_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  def index
+    headers = {
+      "x-rapidapi-host": "joke3.p.rapidapi.com",
+      "x-rapidapi-key": "285c319c04mshb039dd6377ad95cp13cb02jsnf523c698a5dd"
+    }
+    response = RestClient.get("https://joke3.p.rapidapi.com/v1/joke", headers)
+    parsed_response = JSON.parse(response.body)
```
Comment (tochman) [inline app/controllers/api/v1/joke_getter_controller.rb]: Please extract this functionality to a service as I demonstrated in my "Consume API" talk/demo

File: config/routes.rb [hunk 3]
```diff
@@ -1,2 +1,7 @@
 Rails.application.routes.draw do
+  namespace :api do
+    namespace :v1 do
+      resources :joke_getter, only: [:index]
```
Comment (tochman) [inline config/routes.rb]: change this to `:jokes` the `_getter` name is as un-REESTful as it gets.

File: spec/requests/api/v1/joke_getter/index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  it 'is expected to return 200 response status' do
+    expect(response.status).to eq 200
+  end
+
+  it 'is expected to return 1 joke' do
+    expect(response_json["joke_getter"]).to be_truthy
+  end
+
+  it 'is expected to return a random joke' do
+    expect(response_json["joke_getter"]).to eq 'Two dyslexics walk into a bra.'
```
Comment (tochman) [inline spec/requests/api/v1/joke_getter/index_spec.rb]: This is failing right now, right? You should be looking at `response_json["joke_getter"]["content"]` But again, thee `_getter` name is not acceptable IMO.

**Review by tochman — CHANGES_REQUESTED** [review]
Not bad but not really ready for merge yet. Please see and address my comments.

File: spec/requests/api/v1/joke_getter/index_spec.rb (hunk 4 above)
Comment (johanperjulius1) [reply spec/requests/api/v1/joke_getter/index_spec.rb]: It is passing! But we will change the name!

File: config/routes.rb (hunk 3 above)
Comment (johanperjulius1) [reply config/routes.rb]: Ok will change it!

File: app/controllers/api/v1/joke_getter_controller.rb (hunk 1 above)
Comment (johanperjulius1) [reply app/controllers/api/v1/joke_getter_controller.rb]: Done

File: config/routes.rb (hunk 3 above)
Comment (johanperjulius1) [reply config/routes.rb]: Done

File: spec/requests/api/v1/joke_getter/index_spec.rb (hunk 4 above)
Comment (johanperjulius1) [reply spec/requests/api/v1/joke_getter/index_spec.rb]: Done

Comment (grconnor) [reply spec/requests/api/v1/joke_getter/index_spec.rb]: Done

File: app/controllers/api/v1/joke_getter_controller.rb (hunk 2 above)
Comment (grconnor) [reply app/controllers/api/v1/joke_getter_controller.rb]: Done aswell

File: spec/fixtures/files/random_joke.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "id": "db09c5d9659d44448c4da0ae5d321e55",
+  "content": "Two dyslexics walk into a bra.",
+  "nsfw": false,
+  "upvotes": 2,
+  "downvotes": 3
+}
```
Comment (emtalen) [inline spec/fixtures/files/random_joke.json:7]: This is actually funny 😆

**Review by emtalen — APPROVED** [review]

**Pull request merged**